File: CS/Form1.cs
using System;
using System.Drawing.Design;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.UserDesigner;
// ...

namespace EUD_CustomProperty {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            XtraReport report = new XtraReport();
            ReportDesignTool designTool = new ReportDesignTool(report);

            ((XtraReport)designTool.Report).Bands.Add(new DetailBand());
            ((XtraReport)designTool.Report).DesignerLoaded += new DesignerLoadedEventHandler(report_DesignerLoaded);
            ((XtraReport)designTool.Report).ShowDesignerDialog();
        }

        void report_DesignerLoaded(object sender, DesignerLoadedEventArgs e) {
            ToolboxItem item = new ToolboxItem(typeof(CustomControl));
            IToolboxService ts = e.DesignerHost.GetService(typeof(IToolboxService)) as IToolboxService;

            ts.AddToolboxItem(item);
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing.Design
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.UserDesigner
' ...

Namespace EUD_CustomProperty
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim report As New XtraReport()

			report.Bands.Add(New DetailBand())

			AddHandler report.DesignerLoaded, AddressOf report_DesignerLoaded

			report.ShowDesignerDialog()
		End Sub

		Private Sub report_DesignerLoaded(ByVal sender As Object, ByVal e As DesignerLoadedEventArgs)
			Dim item As New ToolboxItem(GetType(CustomControl))
			Dim ts As IToolboxService = TryCast(e.DesignerHost.GetService(GetType(IToolboxService)), IToolboxService)

			ts.AddToolboxItem(item)
		End Sub
	End Class
End Namespace